src/wp-includes/functions.php

```diff
@@ -6998,6 +6998,24 @@ function wp_debug_backtrace_summary( $ignore_class = null, $skip_frames = 0, $pr
  * @return int[] Array of IDs not present in the cache.
  */
 function _get_non_cached_ids( $object_ids, $cache_key ) {
+	$object_ids     = array_filter(
+		$object_ids,
+		static function ( $object_id ) {
+			if ( is_int( $object_id ) ) {
+				return true;
+			}
+
+			// allow "16" but not "16.2" without using `filter_var()` / `ctype_digit()`
```

Comment:
```suggestion
			// allow "16" but not "16.2" without using `filter_var()` / `ctype_digit()`
			/*
			 * Accept integers as either an int or a string.
			 *
			 * This treats both `16` and `"16"` as valid for the purposes of the cache key. Other
			 * numeric types and numeric strings are discarded, for example `16.3` and `"16.3"`.
			 *
			 * This could all be achieved with filter_var(), however the filter extension can be removed
			 * in custom builds so WordPress is unable to rely on it.
			 */
```

It looks like the CS is confused by the code, the NOT keyword or both.